### DISCUSSION TITLE
Dimension outlines around the 3D model using SVG

### DISCUSSION THREAD

**Discussion by odin47** [google/model-viewer#4343] [Q&A]
I've been attempting to give the 3D model dimensions. However, the dimension lines that are generated on top of the 3D model are causing me problems. The model should be surrounded by the lines. Please refer to the screenshot below.

![image](https://github.com/google/model-viewer/assets/16409651/77730e42-a267-4c41-a881-818dacf37327)

I have tried using z-index which svg doesn't support. Can any one help me out here

**Comment by danilokrulj** [comment]
Did you see the example https://modelviewer.dev/examples/annotations/#dimensions ?

**Reply by odin47** [reply]
It says _Note that this kind of extended functionality cannot support occlusion in the 3D scene, as it always draws on top of the model._ I want to to change the z-index of few lines based on the distance from the camera. How do I achieve this?

**Reply by elalish** [reply]
Unfortunately I haven't found a way to do this, since the annotations are in a different stacking context than the render, so they're either all in front, or all behind. If you can think of a solution, I'd be open to a PR.

**Reply by elalish** [reply]
We generally just hide the lines entirely that would end up behind, as a reasonable compromise.

**Reply by odin47** [reply]
Thank you. I just used 3D lines to achieve this functionality.
